bin/configure

```diff
@@ -1,115 +1,82 @@
 #!/usr/bin/env ruby
 
-puts ""
-puts `gem install colorize`
-puts ""
-
-puts ""
-puts `gem install activesupport`
-puts ""
-
 require 'optparse'
 require 'colorize'
 require 'active_support'
+require 'yaml'
+require 'fileutils'
 
-options = {}
-OptionParser.new do |opts|
-  opts.banner = "Usage: bin/configure [options]"
-
-  opts.on("-a", "--api", "Create application in API mode") do |api|
-    options[:api] = api
-  end
-end.parse!
-
-def replace_in_file(file, before, after, target_regexp = nil)
-  puts "Replacing in '#{file}'."
-  if target_regexp
-    target_file_content = ""
-    File.open(file).each_line do |l|
-      l.gsub!(before, after) if !!l.match(target_regexp)
-      l if !!l.match(target_regexp)
-      target_file_content += l
-    end
-  else
-    target_file_content = File.open(file).read
-    target_file_content.gsub!(before, after)
-  end
-  File.open(file, "w+") do |f|
-    f.write(target_file_content)
-  end
-end
+def install_necessary_gems!
```

Comment:
Any reason we use `!` for these methods?

Reply:
Just to differ "step" methods from the secondary ones. Not sure if that's right but it helps in reading code for me personally. Can switch them to normal if we follow the Ruby convention to denote that a method can be "dangerous".

Reply:
Leaved bang only for dangerous methods.